reconcile/utils/saasherder.py

```diff
@@ -131,6 +132,12 @@ def _validate_saas_files(self):
                         saas_file_name,
                         environment_name
                     )
+                    # validate upstream not used with commit sha
+                    self._validate_upstream_not_used_with_commit_sha(
```

Comment:
You only log the error here? This is still trying to apply the promotion.

I'd expect you to handle the `False` return path, or otherwise have this method raise an exception...

Reply:
this is handled here: https://github.com/app-sre/qontract-reconcile/blob/b3a8e9bec00ae27203786e243d9a69663a874267/reconcile/saas_file_validator.py#L30-L31